tests/proofs/dexter/dexter-properties.md

```diff
@@ -1174,30 +1194,33 @@ rule [is-valid]:
 
 #### AddLiquidity(Owner, MinLqtMinted, MaxTokensDeposited, Deadline)
 
+The following rule formulates the behaviors of the AddLiquidity() entrypoint.  It states that, when it succeeds, the entrypoint execution updates the three state variables and the XTZ balance of Dexter.  The execution succeeds when the `assert` conditions are satisfied, and fails otherwise.
+
 ```
 rule [add-liquidity]:
 <operations>  ( [ Transaction Sender DEXTER XtzDeposited AddLiquidity(Owner, MinLqtMinted, MaxTokensDeposited, Deadline) ] => OpsEmitted ) ;; _ </operations>
 <xtzPool>     #Mutez(X => X +Int XtzDeposited)      </xtzPool>
 <tokenPool>          T => T +Int TokensDeposited    </tokenPool>
 <lqtTotal>           L => L +Int LqtMinted          </lqtTotal>
-<xtzDexter>   #Mutez(B => B +Int XtzDeposited)      </xtzDexter>
+<xtzDexter>   #Mutez(B => B')                       </xtzDexter>
 <tokenDexter>        D                              </tokenDexter>
 <lqtSupply>          S                              </lqtSupply>
-requires Sender =/=K DEXTER // TODO: double-check if it is guaranteed
 assert   IS_VALID(Deadline)
  andBool TokensDeposited <=Int MaxTokensDeposited
  andBool LqtMinted       >=Int MinLqtMinted
 ensures  TokensDeposited ==Int XtzDeposited *Int T up/Int X
  andBool LqtMinted       ==Int XtzDeposited *Int L   /Int X
  andBool OpsEmitted ==K [ Transaction DEXTER TOKEN 0 Transfer(Sender, DEXTER, TokensDeposited) ]
                      ;; [ Transaction DEXTER LQT   0 Mint(Owner, LqtMinted) ]
+ andBool Sender =/=K DEXTER impliesBool B' ==Int B +Int XtzDeposited
+ andBool Sender  ==K DEXTER impliesBool B' ==Int B
```

Comment:
Is it possible for Dexter to call AddLiquidity on itself? This seems inconsistent with comments made elsewhere.

Reply:
no it is not possible based on our assumption for the tezos execution model, but i'd like to make this rule to be independent from that.